### PR TITLE
Documentation Updates

### PR DESCRIPTION
**Pull request by unityjeffrey**
Hi all,

I went through the documentation.  At a high level:

- Updated the main README to include the survey link
- Updated the Background-Unity to include links to Prefabs since we reference them extensively with SO Brains
- Cleaning up the Basic Guide to include screenshots and clear up the instructions from the POV of a first time user
- Minor fix in the FAQ
- Cleaned up language in the Getting Started with 3D ball
- Small typos in Imitation Learning
- Minor update to the Glossary

Let me know any comments and happy to change as needed.

Jeff

### CODE REVIEW

**Comment by vincentpierre** [comment]
The update of the Readme with the survey link will cause merge issues that will need to be resolved. We can cherry pick the commit to make it easier later or we can merge anyway and resolve later.

File: docs/Basic-Guide.md [hunk 1]
```diff
@@ -42,33 +42,42 @@ TensorFlow files in the Project window under **Assets** > **ML-Agents** >
 to the `Scripting Define Symbols` for each type of device you want to use 
 (**`PC, Mac and Linux Standalone`**, **`iOS`** or **`Android`**).
 
-![Project Settings](images/project-settings.png)
+   ![Project Settings](images/project-settings.png)
```
Comment (xiaomaogy) [inline docs/Basic-Guide.md]: What are these spaces for?

Comment (unityjeffrey) [reply docs/Basic-Guide.md]: makes the image aligned to the bullet points.  otherwise, it looks odd sitting off to the left

Comment (xiaomaogy) [reply docs/Basic-Guide.md]: Oh I see.

File: docs/Basic-Guide.md [hunk 2]
```diff
@@ -84,30 +93,26 @@ More information and documentation is provided in the
...
+
+   **Note**: The Unity prefab system will modify all instances of the agent properties in your scene.  If the agent does not synchronize automatically with the prefab, you can hit the Revert button in the top of the **Inspector** window.
+
+2. In the **Hierarchy** window, select `Ball3DAcademy`.
+3. In the **Project** window, go to `Assets/ML-Agents/Examples/3DBall/Brains` folder and drag the **3DBallLearning** Brain to the `Brains` property under `Braodcast Hub` in the `Ball3DAcademy` object in the **Inspector** window.  In order to train, make the `Control` checkbox is selected.
```
Comment (xiaomaogy) [inline docs/Basic-Guide.md]: is selected -> selected

File: docs/Basic-Guide.md [hunk 3]
```diff
@@ -84,30 +93,26 @@ More information and documentation is provided in the
...
+
+   **Note:** Assigning a Brain to an agent (dragging a Brain into the `Brain` property of 
 the agent) means that the Brain will be making decision for that agent. Whereas dragging
 a Brain into the Broadcast Hub means that the Brain will be exposed to the Python process.
-The `Control` checkbox means that in addition to being exposed to Python, the Brain will
+The `Control` checkbox means that in addition to being exposed to Python, the agent Brain will
```
Comment (xiaomaogy) [inline docs/Basic-Guide.md]: why "Brain" -> "agent Brain"

File: docs/FAQ.md [hunk 4]
```diff
@@ -71,7 +71,7 @@ There may be a number of possible causes:
   _Solution_: Look into the [log
   files](https://docs.unity3d.com/Manual/LogFiles.html) generated by the Unity
   Environment to figure what error happened.
-# _Cause_: You have assigned HTTP_PROXY and HTTPS_PROXY values in your
+ _Cause_: You have assigned HTTP_PROXY and HTTPS_PROXY values in your
```
Comment (xiaomaogy) [inline docs/FAQ.md]: _Cause_: -> * _Cause_:

File: docs/Basic-Guide.md (hunk 2 above)
Comment (ervteng) [reply docs/Basic-Guide.md]: Maybe it was "make _sure_ the `Control` checkbox is selected?

File: docs/Getting-Started-with-Balance-Ball.md [hunk 5]
```diff
@@ -88,28 +88,24 @@ environment around the Agents.
...
-You can create Brain objects by selecting `Assets -> 
-Create -> ML-Agents -> Brain`. There are 3 kinds of Brains :
-The **Learning Brain** is a Brain that uses a Neural Network to take decisions.
-When the Brain is checked as `Control` in the Academy **Broadcast Hub**, the 
-external process will be taking decisions for the agents
-and generate a neural network when the training is over. You can also use the
+You can create new Brain assets by selecting `Assets -> 
+Create -> ML-Agents -> Brain`. There are 3 types of Brains. 
+The **Learning Brain** is a Brain that uses a trained neural network to make decisions.
+When the `Control` box is checked in the Brains property under the **Broadcast Hub** in the Academy, the external process will  take over decision making for the agents
```
Comment (xiaomaogy) [inline docs/Getting-Started-with-Balance-Ball.md]: Maybe change this to:

When the `Control` box is checked in the Brains property under the **Broadcast Hub** in the Academy, the external process with a learning neural network will take over decision making.

File: docs/FAQ.md (hunk 4 above)
Comment (vincentpierre) [reply docs/FAQ.md]: Agreed, if not the indentation is wrong

File: docs/Basic-Guide.md (hunk 2 above)
Comment (unityjeffrey) [reply docs/Basic-Guide.md]: should be make sure

**Pull request merged**